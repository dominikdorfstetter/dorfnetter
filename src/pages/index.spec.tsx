import renderer from "react-test-renderer";
import React from 'react';
import {IndexPageProps} from './index';
import Index from './index';
import {StaticQuery} from 'gatsby';

const props = {
    data: {
        site: {
            siteMetadata: {
                title: 'app'
            }
        }
    }
} as IndexPageProps;

describe('Index', () => {

    beforeEach(() => {
        // @ts-ignore
        (StaticQuery as any).mockImplementationOnce(({ render }) =>
            render(props.data)
        )
    })

    it("renders correctly", () => {
        const tree = renderer.create(<Index data={props.data} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
