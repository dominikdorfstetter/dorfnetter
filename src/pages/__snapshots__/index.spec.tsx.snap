// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
Array [
  <header>
    <img
      alt="logo"
      className="dorfnetter-logo displayed"
      height="60"
      src="test-file-stub"
      width="auto"
    />
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "marginBottom": "60px",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <p
      className="lead paragraph"
    >
      ðŸ‘‹, I am 
      <span
        className="highlight"
      >
        Dominik
      </span>
      , I am a 31 years old programmer from Vienna.
      <br />
      With a ðŸ’™ for OpenSource.
      <br />
      Angular, Flutter, NestJS, Spring-Boot, Apache Kafka.
    </p>
    <hr />
    <p
      className="lead paragraph social"
    >
      <span
        className="social-container"
      >
        <a
          className="button one"
          href="https://github.com/dominikdorfstetter"
          target="_blank"
        >
          Github
        </a>
        <span
          className="spacer"
        />
        <a
          className="button one"
          href="https://twitter.com/goto_shell"
          target="_blank"
        >
          Twitter
        </a>
        <span
          className="spacer"
        />
        <a
          className="button one"
          href="https://keybase.io/dorfstetter"
          target="_blank"
        >
          Keybase
        </a>
      </span>
    </p>
    <hr />
    <h2>
      Projects
    </h2>
    <p
      className="lead"
    >
      Happy Dogs Proto - A map for dog owners in Vienna. 
      <a
        className="button one"
        href="/happy_dogs/"
        target="_blank"
      >
        Visit
      </a>
    </p>
  </div>,
  <footer
    className="footer mt-auto py-3"
  >
    <div
      className="container footer-text"
    >
      Dorfstetter Dominik 2020 | 
      <a
        aria-current="page"
        className="footer--active"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        Home
      </a>
       |Â 
      <a
        href="/impressum/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Impressum
      </a>
    </div>
  </footer>,
]
`;
